modules/flowable-spring-boot/flowable-spring-boot-starters/flowable-spring-boot-starter-basic/src/test/java/org/flowable/test/spring/boot/cmmn/CmmnEngineAutoConfigurationTest.java

```diff
@@ -0,0 +1,134 @@
+import org.flowable.cmmn.engine.CmmnEngine;
+import org.flowable.engine.ProcessEngine;
+import org.flowable.engine.ProcessEngineConfiguration;
+import org.flowable.engine.impl.util.EngineServiceUtil;
+import org.flowable.spring.boot.FlowableTransactionAutoConfiguration;
+import org.flowable.spring.boot.ProcessEngineAutoConfiguration;
+import org.flowable.spring.boot.cmmn.CmmnEngineAutoConfiguration;
+import org.flowable.spring.boot.cmmn.CmmnEngineServicesAutoConfiguration;
+import org.flowable.spring.boot.idm.IdmEngineAutoConfiguration;
+import org.flowable.spring.boot.idm.IdmEngineServicesAutoConfiguration;
+import org.junit.Test;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceTransactionManagerAutoConfiguration;
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class CmmnEngineAutoConfigurationTest {
+
+    @Test
+    public void standaloneCmmnEngineWithBasicDataSource() {
+        AnnotationConfigApplicationContext context = this.context(
+            DataSourceAutoConfiguration.class,
+            DataSourceTransactionManagerAutoConfiguration.class,
+            IdmEngineAutoConfiguration.class,
+            CmmnEngineServicesAutoConfiguration.class,
+            CmmnEngineAutoConfiguration.class,
+            IdmEngineServicesAutoConfiguration.class
+        );
+
+        CmmnEngine cmmnEngine = context.getBean(CmmnEngine.class);
+        assertThat(cmmnEngine).as("Cmmn engine").isNotNull();
+
+        assertAllServicesPresent(context, cmmnEngine);
+
+        assertAutoDeployment(context);
+    }
+
+    private void assertAllServicesPresent(AnnotationConfigApplicationContext context, CmmnEngine cmmnEngine) {
+        List<Method> methods = Stream.of(CmmnEngine.class.getDeclaredMethods())
+            .filter(method -> !(method.getName().equals("close") || method.getName().equals("getName"))).collect(Collectors.toList());
+
+        assertThat(methods).allSatisfy(method -> {
+            try {
+                assertThat(context.getBean(method.getReturnType())).as(method.getReturnType() + " bean").isEqualTo(method.invoke(cmmnEngine));
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                fail("Failed to invoke method " + method, e);
+            }
+        });
+    }
+
+    private void assertAutoDeployment(AnnotationConfigApplicationContext context) {
+        CmmnRepositoryService repositoryService = context.getBean(CmmnRepositoryService.class);
+
+        List<CaseDefinition> caseDefinitions = repositoryService.createCaseDefinitionQuery().orderByCaseDefinitionName().asc().list();
+        assertThat(caseDefinitions)
+            .extracting(CaseDefinition::getKey)
+            .containsExactly("case1", "case2", "case3", "case4");
```

Comment:
This is the reason why this fails. For some reason the order is different on Travis. I can change the keys, but it is strange why Travis behaves like this. When I run the test locally it is green